tests/Behat/Context/FeatureContext.php

```diff
@@ -0,0 +1,153 @@
+<?php
+namespace DNADesign\Elemental\Tests\Behat\Context;
+
+use Behat\Mink\Element\NodeElement;
+use SilverStripe\BehatExtension\Context\SilverStripeContext;
+
+if (!class_exists(SilverStripeContext::class)) {
+    return;
+}
+class FeatureContext extends SilverStripeContext
+{
+    /**
+     * @Then I should see a list of blocks
+     */
+    public function iShouldSeeAListOfBlocks()
+    {
+        assertNotEmpty($this->getBlocks());
+    }
+
+    /**
+     * @Then I should see block :position
+     */
+    public function iShouldSeeBlock($position)
+    {
+        assertNotNull($this->getSpecificBlock($position));
+        // assert it is actually rendered
+    }
+
+    /**
+     * @When I click on block :position
+     */
+    public function iClickOnBlock($position)
+    {
+        $block = $this->getSpecificBlock($position);
+        assertNotNull($block, 'Block ' . $position . ' was not found in the page.');
+        $block->click();
+    }
+
+    /**
+     * @When I click on the delete button for block :position
+     */
+    public function iClickOnTheDeleteButtonForBlock($position)
+    {
+        $block = $this->getSpecificBlock($position);
+        assertNotNull($block, 'Block ' . $position . ' was not found in the page.');
+
+        $button = $block->find('css', '.font-icon-trash-bin');
+        assertNotNull($button, 'Delete button not found');
+        $button->click();
+    }
+
+    /**
+     * @Then I should see :text as the icon for block :position
+     */
+    public function iShouldSeeAsTheIconForBlock($text, $position)
+    {
+        $block = $this->getSpecificBlock($position);
+        $icon = $block->find('css', '.element-editor-header__icon-container .i');
+        assertTrue($icon->hasClass($text));
+    }
+
+    /**
+     * @Then I should see :text as the title for block :position
+     */
+    public function iShouldSeeAsTheTitleForBlock($text, $position)
+    {
+        $block = $this->getSpecificBlock($position);
+        $title = $block->find('css', '.element-editor-header__title');
+        assertEquals($title->getText(), $text);
+    }
+
+    /**
+     * @Then /^I should (not |)see the edit link for block :position/
+     *
+     * @param string $text
+     * @param string $position
+     */
+    public function iShouldSeeEditLinkForBlock($text, $position)
+    {
+        $block = $this->getSpecificBlock($position);
+        $editLink = $block->find('css', '.element-editor-header__actions-container .a');
+        assertTrue($icon->hasClass($text));
+    }
+
+
+    /**
+     * @Then I should see the delete button for block :position
+     *
+     * @param string $text
+     * @param string $position
+     */
+    public function iShouldSeeDeleteButtonForBlock($position)
+    {
+        $block = $this->getSpecificBlock($position);
+        $button = $block->findAll('css', 'Delete');
+        assertNotNull($button, 'Delete button not found');
```

Comment:
Do you think we could combine some logic in this step with `iClickOnTheDeleteButtonForBlock`?